src/com/android/calendar/month/MonthWeekEventsView.java

```diff
@@ -785,152 +753,912 @@ protected int addChipOutline(FloatRef lines, int count, int x, int y) {
+            for (ArrayList<Event> dayEvents : mEvents) {
+                mFormattedEvents.add(prepareFormattedEventDay(dayEvents, lastDayEvents, day, daysInWeek));
+                ++day;
+                lastDayEvents = (dayEvents == null) ? new ArrayList<Event>() : dayEvents;
+            }
+        }
+    }
+
     /**
-     * Attempts to draw the given event. Returns the y for the next event or the
-     * original y if the event will not fit. An event is considered to not fit
-     * if the event and its extras won't fit or if there are more events and the
-     * more events line would not fit after drawing this event.
+     * Takes care of laying events out vertically.
+     * Vertical layout:
+     *   (top of box)
+     * a. EVENT_Y_OFFSET_LANDSCAPE or portrait equivalent
+     * b. Event title: mEventHeight for a normal event, + 2xBORDER_SPACE for all-day event
+     * c. [optional] Time range (mExtrasHeight)
+     * d. EVENT_LINE_PADDING
      *
-     * @param canvas the canvas to draw on
-     * @param event the event to draw
-     * @param x the top left corner for this event's color chip
-     * @param y the top left corner for this event's color chip
-     * @param rightEdge the rightmost point we're allowed to draw on (exclusive)
-     * @param moreEvents indicates whether additional events will follow this one
-     * @param showTimes if set, a second line with a time range will be displayed for non-all-day
-     *   events
-     * @param doDraw if set, do the actual drawing; otherwise this just computes the height
-     *   and returns
-     * @return the y for the next event or the original y if it won't fit
+     * Repeat (b,c,d) as needed and space allows.  If we have more events than fit, we need
+     * to leave room for something like "+2" at the bottom:
+     *
+     * e. "+ more" line (mExtrasHeight)
+     *
+     * f. EVENT_BOTTOM_PADDING (overlaps EVENT_LINE_PADDING)
+     *   (bottom of box)
      */
-    protected int drawEvent(Canvas canvas, Event event, int x, int y, int rightEdge,
-            boolean moreEvents, boolean showTimes, boolean doDraw) {
-        /*
-         * Vertical layout:
-         *   (top of box)
-         * a. EVENT_Y_OFFSET_LANDSCAPE or portrait equivalent
-         * b. Event title: mEventHeight for a normal event, + 2xBORDER_SPACE for all-day event
-         * c. [optional] Time range (mExtrasHeight)
-         * d. EVENT_LINE_PADDING
-         *
-         * Repeat (b,c,d) as needed and space allows.  If we have more events than fit, we need
-         * to leave room for something like "+2" at the bottom:
-         *
-         * e. "+ more" line (mExtrasHeight)
-         *
-         * f. EVENT_BOTTOM_PADDING (overlaps EVENT_LINE_PADDING)
-         *   (bottom of box)
+    protected class DayEventFormatter {
+        private ArrayList<FormattedEventBase> mEventDay;
+        private boolean mShowTimes;
+        private int mDay;
+        //info
```

Comment:
What should the comment mean?

Reply:
comment changed to a more meaningful one